[scd] Add hash lock option (#1520)
interuss/dss: docs/operations/performances.md

![](../assets/perfs_scd_lock_notoverlapping.png)
*Non-overlapping requests. Notice the reduction of performance on the right, with a single lock.*

> ## The SCD hash lock option

Reviewer: Clarify how those options are(n't) cumulative: only one may be used at a time (I believe?)

!!! danger
     All DSS instances in a DSS pool must use the same value for this option. Mixing will result in dramatically lower performance.

     You can use the `/aux/v1/configuration/global_options` endpoint to retrieve the current value for a specifc DSS instance.

!!! danger
    This option requires migration 3.4.1 for CockroachDB and 1.1.1 for Yugabyte to be effective

> Similar to the SCD global lock option, but use lock based on hashed cells (with 65k locks).

Reviewer: How was this value chosen? 
Can it be changed? (no)

![](../assets/perfs_scd_lock_notoverlapping.png)
*Non-overlapping requests. Notice the reduction of performance on the right, with a single lock.*

## The SCD hash lock option

!!! danger
     All DSS instances in a DSS pool must use the same value for this option. Mixing will result in dramatically lower performance.

>      You can use the `/aux/v1/configuration/scd_lock_mode` endpoint to retrive the current values of locks for a specifc DSS instance.

Reviewer 2: ```suggestion
     You can use the `/aux/v1/configuration/scd_lock_mode` endpoint to retrieve the current values of locks for a specifc DSS instance.
```

!!! danger
    This option requires migration 3.4.1 for CockroachDB and 1.1.1 for Yugabyte to be effective

> Similar to the previous option, but use lock based on hashed cells (with 65k locks).

Reviewer 2: ```suggestion
Similar to the SCD global lock option, but use lock based on hashed cells (with 65k locks).
```

> This should be better than global lock, as long as cells used don't colide and limit number of entries in the database.

Reviewer 2: ```suggestion
This should be better than global lock, as long as cells used don't collide and limit number of entries in the database.
```
